gnomad_qc/v4/variant_qc/evaluation.py

```diff
@@ -20,91 +19,104 @@
+    if vqsr:
+        struct_expr = struct_expr.annotate(
             score=ht.info.AS_VQSLOD,
             positive_train_site=ht.info.POSITIVE_TRAIN_SITE,
             negative_train_site=ht.info.NEGATIVE_TRAIN_SITE,
             AS_culprit=ht.info.AS_culprit,
         )
-
-        if hgdp_tgp_subset:
-            allele_data_ht = allele_data.ht()
-            ht = ht.annotate(**allele_data_ht[ht.key].allele_data)
-        else:
-            # Remove all variants with an undefined ac_raw because ac_raw was calculated on the high quality samples only
-            # and VQSR was run before sample filtering
-            ht = ht.filter(hl.is_defined(ht.ac_raw))
-
     else:
-        if hgdp_tgp_subset:
-            raise ValueError(
-                "The hgdp_tgp_subset option is only relevant for VQSR models in all v3"
-                " releases."
-            )
-
-        ht = get_rf_result(model_id=model_id).ht()
-        ht = ht.annotate(
-            info=info_ht[ht.key].info,
+        struct_expr = struct_expr.annotate(
+            score=ht.rf_probability["TP"],
             positive_train_site=ht.tp,
             negative_train_site=ht.fp,
-            score=ht.rf_probability["TP"],
         )
 
-    ht = ht.filter(
-        ~info_ht[ht.key].AS_lowqual
-        & ~hl.is_defined(telomeres_and_centromeres.ht()[ht.locus])
-    )
-    ht_non_lcr = filter_low_conf_regions(
+    ht = ht.annotate(**struct_expr, non_lcr=hl.is_defined(non_lcr_ht[ht.key]))
+    ht = ht.checkpoint(new_temp_file("pre_bin", "ht"), overwrite=True)
+    bin_ht = create_binned_ht(
         ht,
-        filter_lcr=True,
-        # TODO: Uncomment when we have decoy path
-        filter_decoy=False,  # True,
-        filter_segdup=True,
+        n_bins,
+        add_substrat={
+            "non_lcr": ht.non_lcr,
+            "in_calling_intervals": ht.in_calling_intervals,
+            "pass_interval": ht.pass_interval_qc,
+        },
     )
-    ht = ht.annotate(non_lcr=hl.is_defined(ht_non_lcr[ht.key]))
-    bin_ht = create_binned_ht(ht, n_bins, add_substrat={"non_lcr": ht.non_lcr})
 
     return bin_ht
 
 
-def create_aggregated_bin_ht(model_id: str) -> hl.Table:
+def score_bin_validity_check(ht: hl.Table) -> None:
+    """
+    Check that the bin scoring looks correct by printing so aggregate numbers.
+
+    :param ht: Table with bin annotations.
+    :return: None
+    """
+    print(
+        ht.aggregate(
+            hl.struct(
+                was_biallelic=hl.agg.counter(~ht.was_split),
+                has_biallelic_rank=hl.agg.counter(hl.is_defined(ht.biallelic_bin)),
+                was_singleton=hl.agg.counter(ht.singleton),
+                has_singleton_rank=hl.agg.counter(hl.is_defined(ht.singleton_bin)),
+                was_biallelic_singleton=hl.agg.counter(ht.singleton & ~ht.was_split),
+                has_biallelic_singleton_rank=hl.agg.counter(
+                    hl.is_defined(ht.biallelic_singleton_bin)
```

Comment:
this fails for me on VQSR - is there a way to check if these exist before printing?

Reply:
that's odd, with the current gnomad_main branch, `biallelic_singleton_bin` should get added. The one I made for you last night has it I think:

```2023-10-03 04:47:37.425 Hail: INFO: wrote table with 199825814 rows in 5000 partitions to gs://gnomad/v4.0/variant_qc/exomes/score_bins/gnomad.exomes.v4.0.vqsr_missing_raw_ts_ss.bins.ht
INFO (variant_qc_evaluation 149): Found the following variant counts:0 + 2) / 2]
 Struct(non_lcr_adj_bin={'indel': 5648220, 'snv': 61084429},
       non_lcr_singleton_bin={'indel': 4146644, 'snv': 18614016},
       adj_biallelic_bin={'indel': 475440, 'snv': 12718407},
       pass_interval_biallelic_singleton_bin={'indel': 74765, 'snv': 1330509},
       in_calling_intervals_adj_bin={'indel': 3129737, 'snv': 33977126},
       pass_interval_singleton_bin={'indel': 329417, 'snv': 3100511},
       in_calling_intervals_adj_biallelic_bin={'indel': 300245, 'snv': 8405698},
       pass_interval_adj_biallelic_singleton_bin={'indel': 55535,
                                                  'snv': 1026837},
       biallelic_singleton_bin={'indel': 396750, 'snv': 5449291},
       adj_singleton_bin={'indel': 2170485, 'snv': 12674815},
       in_calling_intervals_singleton_bin={'indel': 1833293, 'snv': 10889633},
       bin={'indel': 18391422, 'snv': 181434392},
       non_lcr_biallelic_singleton_bin={'indel': 352793, 'snv': 4957593},
       pass_interval_adj_bin={'indel': 522654, 'snv': 7230032},
       pass_interval_adj_biallelic_bin={'indel': 104580, 'snv': 2807249},
       pass_interval_bin={'indel': 712751, 'snv': 8844620},
       pass_interval_adj_singleton_bin={'indel': 232489, 'snv': 2379877},
       non_lcr_biallelic_bin={'indel': 999272, 'snv': 29232632},
       pass_interval_biallelic_bin={'indel': 131718, 'snv': 3309501},
       in_calling_intervals_adj_biallelic_singleton_bin={'indel': 155422,
                                                         'snv': 2747607},
       non_lcr_adj_biallelic_singleton_bin={'indel': 202142, 'snv': 3331313},
       adj_bin={'indel': 7488903, 'snv': 68926679},
       in_calling_intervals_bin={'indel': 5039399, 'snv': 50154719},
       biallelic_bin={'indel': 1148563, 'snv': 32101417},
       singleton_bin={'indel': 5130500, 'snv': 20673919},
       adj_biallelic_singleton_bin={'indel': 221036, 'snv': 3613686},
       in_calling_intervals_adj_singleton_bin={'indel': 1126193, 'snv': 8186769},
       in_calling_intervals_biallelic_bin={'indel': 390645, 'snv': 10637105},
       non_lcr_bin={'indel': 13859410, 'snv': 161758550},
       non_lcr_adj_singleton_bin={'indel': 1870599, 'snv': 11552538},
       in_calling_intervals_biallelic_singleton_bin={'indel': 211058,
                                                     'snv': 3589479},
       non_lcr_adj_biallelic_bin={'indel': 428093, 'snv': 11562226})
```

Reply:
oh, I actually didn't run it with this flag on last night , I didn't know it got fixed. just checked with a quick test and it does look fixed!